WebUI improvements

Chocobo1 opened the pull request:
* WebUI: remove child elements directly
* WebUI: use correct property for selecting child elements
  `firstChild` will select the first `Node` which is often not intended (it should be `Element` instead).
* WebUI: use idiomatic string methods
* WebUI: use proper event for handling text changes
* WebUI: fix wrong event property
* WebUI: fix invalid style
  `initial` isn't applicable to `borderLeft`/`borderRight`.
* WebUI: fix checkbox initialization
  Previously the checkbox had all options checked regardless of the stored setting.
* WebUI: revise Edit Category dialog button text
  The generic "OK" is suitable for more scenarios.
* WebUI: locate element faster

thalieht commented:
Since i have no idea what almost all of these changes do i started testing random stuff.

Caused by this PR:
Filters side panel: tags and trackers text are missing, both light and dark theme.

Not caused by this PR:
Remove torrent dialog: Cancel has no effect and remove doesn't close the dialog
Rename files dialog: Close button has no effect
Edit category dialog: Button is called "Add", add what? Shouldn't it be "Save" or "Ok" like in GUI?

Chocobo1 commented:
>Since i have no idea what almost all of these changes do i started testing random stuff.

Appreciated the testing!

>Filters side panel: tags and trackers text are missing, both light and dark theme.

My fault, fixed.

>Remove torrent dialog: Cancel has no effect and remove doesn't close the dialog
>Rename files dialog: Close button has no effect

I was looking into it too earlier. Those are regression from #21804. I've already fixed them in my unpublished branch. I'll submit a PR later.

>Edit category dialog: Button is called "Add", add what? Shouldn't it be "Save" or "Ok" like in GUI?

I've changed it to "OK".
